source/common/network/udp_listener_impl.cc

```diff
@@ -112,56 +124,83 @@ void UdpListenerImpl::handleReadCallback() {
 
     Address::InstanceConstSharedPtr peer_address;
 
-    // TODO(conqerAtApple): Current implementation of Address::addressFromSockAddr
-    // cannot be used here unfortunately. This should belong in Address namespace.
-    switch (addr.ss_family) {
-    case AF_INET: {
-      const struct sockaddr_in* sin = reinterpret_cast<const struct sockaddr_in*>(&addr);
-      ASSERT(AF_INET == sin->sin_family);
-      peer_address = std::make_shared<Address::Ipv4Instance>(sin);
-
-      break;
-    }
-    case AF_INET6: {
-      const struct sockaddr_in6* sin6 = reinterpret_cast<const struct sockaddr_in6*>(&addr);
-      ASSERT(AF_INET6 == sin6->sin6_family);
-      if (IN6_IS_ADDR_V4MAPPED(&sin6->sin6_addr)) {
-#if defined(__APPLE__)
-        struct sockaddr_in sin = {
-            {}, AF_INET, sin6->sin6_port, {sin6->sin6_addr.__u6_addr.__u6_addr32[3]}, {}};
-#else
-        struct sockaddr_in sin = {AF_INET, sin6->sin6_port, {sin6->sin6_addr.s6_addr32[3]}, {}};
-#endif
-        peer_address = std::make_shared<Address::Ipv4Instance>(&sin);
-      } else {
-        peer_address = std::make_shared<Address::Ipv6Instance>(*sin6, true);
-      }
-
-      break;
-    }
-
-    default:
-      RELEASE_ASSERT(false,
-                     fmt::format("Unsupported address family: {}, local address: {}, receive size: "
-                                 "{}, address length: {}",
-                                 addr.ss_family, local_address->asString(), recv_result.result_.rc_,
-                                 addr_len));
-      break;
+    try {
+      peer_address = Address::addressFromSockAddr(
+          addr, addr_len, local_address->ip()->version() == Address::IpVersion::v6);
+    } catch (const EnvoyException&) {
+      // Intentional no-op. The assert should fail below
```

Comment:
Can this happen?

Reply:
I do see that the addressFromSockAddr can throw an exception - yes.